skyportal/handlers/api/annotation.py

```diff
@@ -30,18 +49,64 @@ def get(self, annotation_id):
               application/json:
                 schema: Error
         """
-        annotation = Annotation.get_if_accessible_by(
-            annotation_id, self.current_user, raise_if_none=True
-        )
+
+        try:
+            annotation_id = int(annotation_id)
+        except (TypeError, ValueError):
+            return self.error("Must provide a valid annotation ID. ")
+        # the default is to annotate an object
+        if associated_resource_type.lower() in (
+            "object",
+            "objects",
+            "source",
+            "sources",
+        ):
+            annotation = Annotation.get_if_accessible_by(
+                annotation_id, self.current_user, raise_if_none=True
+            )
+            annotation_resource_id_str = str(annotation.obj_id)
+        # add more options using elif
+        else:
+            return self.error(
+                f'Unsupported input "{associated_resource_type}" given as "associated_resource_type" argument.'
+            )
+
+        if annotation_resource_id_str != resource_id:
+            return self.error(
+                f'Annotation resource ID does not match resource ID given in path ({resource_id})'
+            )
+
         return self.success(data=annotation)
 
     @permissions(['Annotate'])
-    def post(self):
+    def post(self, associated_resource_type, resource_id, *_):
         """
         ---
         description: Post an annotation
         tags:
           - annotations
+        parameters:
+          - in: path
+            name: associated_resource_type
+            required: true
+            schema:
+              type: string
+            description: |
+               What underlying data the annotation is on:
+               an "object" (default), or a "spectrum".
+          - in: path
+            name: resource_id
+            required: true
+            schema:
+              type: string
+            description: |
+               The ID of the underlying data.
+               This would be a string for an object ID
+               or an integer for other data types like spectrum.
+               The object pointed to by this input must be a valid
+               object or other data type (like spectrum) that
+               can be annotated on by the user/token.
+               It must match the data given in the request body.
```

Comment:
Why specify it twice?  Should be sufficient to grab it from the URL.

Reply:
Just for consistency. But if you don't think that matters much then we can remove it.

Reply:
Let's only specify it in one place.

Reply:
OK